### PR TITLE
Ayatamimi fastapi task

### PR DESCRIPTION
**Pull request by ayanourr**
This pull request introduces a Task Management API using FastAPI. The API provides functionality to manage tasks with operations including creation, updating, retrieval, and deletion. Each task has attributes such as title, description, due date, status, and optional priority.

### CODE REVIEW

File: aya_tamimi/utils/task_utils.py
Comment (FaresAbuIram) [inline aya_tamimi/utils/task_utils.py:1]: I think we don't need this since we are using the base model

**Review by FaresAbuIram — CHANGES_REQUESTED** [review]

File: aya_tamimi/models/basemodels.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from pydantic import BaseModel
+from models.enums import PriorityEnum,StatusEnum
+from datetime import date
+
+
+# this class for creation needs only
+class Create_Task_Basemodel(BaseModel):
+    title: str
+    description: str
+    due_date: date
```
Comment (FaresAbuIram) [inline aya_tamimi/models/basemodels.py:10]: here you can check if the data is in YYYY:MM: DD format.

Comment (ayanourr) [reply aya_tamimi/models/basemodels.py:10]: The 'date' type expects the date to be only in YYYY-MM-DD format, so it automatically handles the validation

Comment (FaresAbuIram) [reply aya_tamimi/models/basemodels.py:10]: oh, I see

**Review by FaresAbuIram — APPROVED** [review]

File: aya_tamimi/task_endpoints.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+
+# Update the details of an existing task by its ID.
+@app.put("/tasks/{task_id}")
+def update_task(task_id: uuid.UUID, task_update: Update_Task_Basemodel):
+    try:
+        task_to_update = find_task_by_id(task_id)
+        if not task_to_update:
+            raise HTTPException(status_code=404, detail="Task not found")
+        # Update the task attributes if they are provided
```
Comment (Momen-G-Ar) [inline aya_tamimi/task_endpoints.py:30]: What happens if the attributes are more? 10? 20? 
are u sure to use this way? 
let us discuss the problem: you do the same operations for all properties for an object, I think it is better to use a loop to iterate over all of them and handle these operations.

**Review by Momen-G-Ar — CHANGES_REQUESTED** [review]
Other than the edit required: Your code is great. Thank you

File: aya_tamimi/models/basemodels.py (hunk 1 above)
Comment (FaresAbuIram) [reply aya_tamimi/models/basemodels.py:10]: @JbreenJboor1212 the first validation should be enough and always should be more options. 

Thank you

Comment (ayanourr) [reply aya_tamimi/models/basemodels.py:10]: Appreciate the insights! I'll keep it in mind for future work. 
Thanks for the collaboration!

File: aya_tamimi/task_endpoints.py (hunk 2 above)
Comment (ayanourr) [reply aya_tamimi/task_endpoints.py:30]: I’ve switched to a loop for dynamic updates to keep things scalable. 
Appreciate the suggestion made a big difference!

**Review by Momen-G-Ar — APPROVED** [review]

**Pull request merged**